draft-ietf-rtcweb-jsep.xml

```diff
@@ -1799,9 +1799,27 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
 
           <t>For media m= sections, JSEP implementations MUST support
           the "UDP/TLS/RTP/SAVPF" profile specified in
-          <xref target="RFC5764"></xref>, and MUST indicate this
-          profile for each media m= line they produce in an offer. For
-          data m= sections, implementations MUST support the
+          <xref target="RFC5764"></xref>. Implementations MUST indicate this
+          profile for each media m= line they produce in an offer unless the
+          media section contains only TCP candidates; if all candidates use TCP as a
+          transport, implementations MUST indicate "TCP/DTLS/RTP/SAVPF," to
+          allow for compatibility with non-WebRTC implementations of
+          <xref target="I-D.ietf-mmusic-ice-sip-sdp"></xref>.</t>
+          
+            <t><list style="hanging">
+              <t hangText="Note:">Current implementations of JSEP do not update m= or c= lines
```

Comment:
Currently it is fairly simple to use WebRTC JavaScript API to implement something which is RFC 5245 compatible, including generation of the extra "clean up" offer. Browser does not have to force the "clean up" offer. If an application needs such offer, this offer can generated by an application when ICE nomination is complete. What we are trying to make sure here is that SDP generated in such offer is compatible with RFC 5245, i.e. includes only default ICE candidates and have address and protocol information in c=m= line that match the default candidate. What we are trying to avoid is the need to SDP patching to keep this update offer compatible with legacy RFC 5245, which only happens when TCP candidate is nominated.